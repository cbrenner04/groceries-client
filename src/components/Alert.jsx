import React from 'react';
import PropTypes from 'prop-types';
import { Alert as BootstrapAlert } from 'react-bootstrap';

function Alert(props) {
  if (!props.errors && !props.success) return null;
  return (
    <BootstrapAlert variant={props.errors ? 'danger' : 'success'} onClose={props.handleDismiss} dismissible>
      {props.errors || props.success}
    </BootstrapAlert>
  );
}

Alert.propTypes = {
  errors: PropTypes.string,
  success: PropTypes.string,
  handleDismiss: PropTypes.func.isRequired,
};

Alert.defaultProps = {
  errors: '',
  success: '',
};

export default Alert;
