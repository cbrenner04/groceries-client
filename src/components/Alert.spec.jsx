import React from 'react';
import Alert from './Alert';
import { render, fireEvent  } from '@testing-library/react'

const handleDismiss = jest.fn();

describe('Alert', () => {
  describe('when error is provided', () => {
    it('renders alert with error class and error message', () => {
      const errorMessage = 'test error message';
      const { getByRole, getByText } = render(<Alert errors={errorMessage} handleDismiss={handleDismiss} />);

      expect(getByRole('alert')).toMatchSnapshot();
      expect(getByRole('alert')).toHaveClass('alert-danger');
      expect(getByText(errorMessage));
    });
  });

  describe('when success is provided', () => {
    it('renders alert with success class and success message', () => {
      const successMessage = 'test success message';
      const { getByRole, getByText } = render(<Alert success={successMessage} handleDismiss={handleDismiss} />);

      expect(getByRole('alert')).toMatchSnapshot();
      expect(getByRole('alert')).toHaveClass('alert-success');
      expect(getByText(successMessage));
    });
  });

  describe('when error or success is not provided', () => {
    it('renders nothing', () => {
      const { container } = render(<Alert handleDismiss={handleDismiss} />);

      expect(container.firstChild).toBeNull();
    });
  });

  describe('when close button is selected', () => {
    it('calls handleDismiss', () => {
      const { getByRole } = render(<Alert errors="test" handleDismiss={handleDismiss} />);
      fireEvent.click(getByRole('button'));

      expect(handleDismiss).toHaveBeenCalled();
    });
  });
});
