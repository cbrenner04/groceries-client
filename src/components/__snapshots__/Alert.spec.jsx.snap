// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert when error is provided renders alert with error class and error message 1`] = `
<div
  class="fade alert alert-danger alert-dismissible show"
  role="alert"
>
  <button
    class="close"
    type="button"
  >
    <span
      aria-hidden="true"
    >
      ×
    </span>
    <span
      class="sr-only"
    >
      Close alert
    </span>
  </button>
  test error message
</div>
`;

exports[`Alert when success is provided renders alert with success class and success message 1`] = `
<div
  class="fade alert alert-success alert-dismissible show"
  role="alert"
>
  <button
    class="close"
    type="button"
  >
    <span
      aria-hidden="true"
    >
      ×
    </span>
    <span
      class="sr-only"
    >
      Close alert
    </span>
  </button>
  test success message
</div>
`;
